src/lib/secret.ts

```diff
@@ -1,22 +1,36 @@
 export class Secret {
   // A fauna key, like `fn1234`.
   key: string;
+  // Do we allow database scope?
+  allowDatabase: boolean;
   // A database scope, like `["foo", "bar"]`
   databaseScope: string[];
 
-  constructor(opts: { key: string; databaseScope?: string[] }) {
+  constructor(opts: {
+    key: string;
+    allowDatabase: boolean;
+    databaseScope?: string[];
+  }) {
     this.key = opts.key;
+    this.allowDatabase = opts.allowDatabase;
     this.databaseScope = opts.databaseScope ?? [];
   }
 
+  static parseFlag(key: string) {
+    if (key.length === 0) {
+      throw new Error("Secret cannot be empty");
+    }
+    return new Secret({ key, allowDatabase: !key.includes(":") });
+  }
+
   static parse(key: string) {
```

Comment:
wondering if we should just have one parse method, could take as a parameter whether or not the key can be scoped.  Could use both of these methods under the hood.  looking at 2 methods parse vs parseFlag may not be clear which to use but if we had an explicit parameter, `allowScopedKey` or something, then it may make it a bit more clear.